db/db_impl_secondary.cc

```diff
@@ -52,12 +58,195 @@ Status DBImplSecondary::Recover(
     default_cf_internal_stats_ = default_cf_handle_->cfd()->internal_stats();
     single_column_family_mode_ =
         versions_->GetColumnFamilySet()->NumberOfColumnFamilies() == 1;
+
+    // Recover from all newer log files than the ones named in the
+    // descriptor (new log files may have been added by the previous
+    // incarnation without registering them in the descriptor).
```

Comment:
`Previous incarnation` is misleading. Previous incarnation of secondary instance does not add WAL.